service/k8sclient/k8sclient.go

```diff
@@ -346,6 +351,34 @@ func init() {
 	}
 
 	pmmClientImage = "percona/pmm-client:" + v.Core().String()
+
+	exists, err := imageExists(context.Background(), pmmClientImage)
+
+	// if !exists or there was an error while checking if the image exists, use dev-latest as default.
+	if !exists || err != nil {
+		pmmClientImage = devLatest
+	}
+}
+
+func imageExists(ctx context.Context, image string) (bool, error) {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		panic(err)
+	}
+	defer cli.Close() //nolint
+
+	reader, err := cli.ImagePull(ctx, image, dockerTypes.ImagePullOptions{})
```

Comment:
Is there a way of checking if the image exists without pulling it? Seems like an unnecessary action.

Reply:
It doesn't really pull. It returns a reader interface which is closed in the line below this without actually downloading the image.

Reply:
ahhh ok great